src/ledger/LedgerManagerImpl.cpp

```diff
@@ -585,11 +586,29 @@ LedgerManagerImpl::loadLastKnownLedger(bool restoreBucketlist)
     auto const& snapshot = mLastClosedLedgerState->getBucketSnapshot();
     mApplyState.compileAllContractsInLedger(snapshot,
                                             latestLedgerHeader->ledgerVersion);
-    mApplyState.populateInMemorySorobanState(snapshot,
-                                             latestLedgerHeader->ledgerVersion);
+    if (populateInMemoryState)
```

Comment:
Since this is a performance optimization for debug commands (I assume), can we also do compilation behind the flag?

Reply:
Sorry, by "compile behind the flag" I mean call the compilation function inside `populateInMemoryState` block, not the BUILD_TEST flag.